Products/CMFPlone/browser/login/login.py

```diff
@@ -205,9 +221,28 @@ def force_password_change(self):
             handler()
 
     def redirect_after_login(self, came_from=None, is_initial_login=False):
+        """Handle redirect after successful login."""
         adapter = queryMultiAdapter((self.context, self.request), IRedirectAfterLogin)
         if adapter:
             came_from = adapter(came_from, is_initial_login)
+        
+        if came_from:
+            try:
+                # Parse URL and get path for existence check
+                parsed_url = parse.urlparse(came_from)
+                path = parsed_url.path.split("/")
+                path = path[1:] if path[0] == '' else path
```

Comment:
You generate your path without leading `/` ... I wonder if this works correctly in VHM environments.

All in all this needs an additional test ... there are also tests which simulates VHM environments. Would make sense here.

Reply:
okay thats something, i am unsure!

Reply:
I've simplified and tested this in (non-)VHM environment and it works in both scenarios. Lets see what jenkins says.

Reply:
Thanks for your efforts... Hopefully it passes!